torchmultimodal/diffusion_labs/models/adm_unet/adm.py

```diff
@@ -378,7 +346,65 @@ def forward(
             if hidden_states:
                 h = torch.cat([h, hidden_states.pop()], dim=1)
             h = block(h, res_conditional_embedding, attn_conditional_embedding)
-        return h
+
+        # If model is predicting variance, then it should be configured to output double the channels as input
+        if self.predict_variance_value:
+            if h.shape[1] != x.shape[1] * 2:
+                raise ValueError(
+                    f"unet is not configured to predict variance values. "
+                    f"Expected output channel dim to be {x.shape[1] * 2}, got {h.shape[1]}"
+                )
+            # Split in half in channel dim
+            prediction, variance_value = torch.chunk(h, 2, dim=1)
+            variance_value = self.variance_value_transform(variance_value)
+            return DiffusionOutput(
+                prediction=prediction,
+                variance_value=variance_value,
+            )
+        else:
+            return DiffusionOutput(
+                prediction=h,
+            )
+
+    def _get_conditional_projections(
```

Comment:
nit I think it's better to have forward as the last method in a nn.Module so it's easier to find.